boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,373 @@
+    GPIO_TypeDef *           fsob_port,
+    uint16_t                 fsob_pin,
+    GPIO_TypeDef *           fsb_port,
+    uint16_t                 fsb_pin,
+    GPIO_TypeDef *           channel0_port,
+    uint16_t                 channel0_pin,
+    GPIO_TypeDef *           channel1_port,
+    uint16_t                 channel1_pin)
+{
+    assert(hspi != NULL);
+
+    struct Efuse_Context *efuse_context = malloc(sizeof(struct Efuse_Context));
+    assert(efuse_context != NULL);
+
+    efuse_context->hspi            = hspi;
+    efuse_context->nss_port        = chip_select_port;
+    efuse_context->nss_pin         = chip_select_pin;
+    efuse_context->fsob_port       = fsob_port;
+    efuse_context->fsob_pin        = fsob_pin;
+    efuse_context->fsb_port        = fsb_port;
+    efuse_context->fsb_pin         = fsb_pin;
+    efuse_context->channel0_port   = channel0_port;
+    efuse_context->channel0_pin    = channel0_pin;
+    efuse_context->channel1_port   = channel1_port;
+    efuse_context->channel1_pin    = channel1_pin;
+    efuse_context->wdin_bit_to_set = true;
+
+    return efuse_context;
+}
+
+void Io_Efuse_Destroy(struct Efuse_Context *e_fuse)
+{
+    free(e_fuse);
+}
+
+void Io_Efuse_Channel0Enable(const struct Efuse_Context *const e_fuse)
+{
+    HAL_GPIO_WritePin(
+        e_fuse->channel0_port, e_fuse->channel0_pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_Channel0Disable(const struct Efuse_Context *const e_fuse)
+{
+    HAL_GPIO_WritePin(
+        e_fuse->channel0_port, e_fuse->channel0_pin, GPIO_PIN_RESET);
+}
+
+void Io_Efuse_Channel1Enable(const struct Efuse_Context *const e_fuse)
+{
+    HAL_GPIO_WritePin(
+        e_fuse->channel1_port, e_fuse->channel1_pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_Channel1Disable(const struct Efuse_Context *const e_fuse)
+{
+    HAL_GPIO_WritePin(
+        e_fuse->channel1_port, e_fuse->channel1_pin, GPIO_PIN_RESET);
+}
+
+ExitCode Io_Efuse_ConfigureChannelMonitoring(
+    uint8_t                     selection,
+    struct Efuse_Context *const e_fuse)
+{
+    uint16_t register_value = 0x0000;
+
+    // Read original content of GCR Register
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_ReadRegister(SI_GCR_ADDR, &register_value, e_fuse));
+
+    CLEAR_BIT(register_value, (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(register_value, (selection & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_GCR_ADDR, register_value, e_fuse));
+
+    return EXIT_CODE_OK;
+}
+
+ExitCode Io_Efuse_ExitFailSafeMode(struct Efuse_Context *const e_fuse)
+{
+    // Set WDIN bit for next write
+    // 1_1_00000_00000_0000
+    e_fuse->wdin_bit_to_set = true;
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_STATR_0_ADDR, 0x0000, e_fuse));
+
+    // Disable the watchdog timer
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_GCR_ADDR, GCR_CONFIG, e_fuse));
+
+    // Check if the the efuse is still in fail-safe mode
```

Comment:
We can use Io_Efuse_IsEfuseInFailsafeMode here

Reply:
Also we probably shouldn't have the AUX1_AUX2 constants in the Io_Efuse.c file

Reply:
haha lol, copy-pasta moment